File: CPP/DsAlgoSTL/stack.cpp
#include <iostream>
#include <stack>
#include <cstring>
using namespace std;

// this program is basic implementation of stack using STL cpp Library
// below mentioned stack functions are discussed

// .push()
// .pop()
// .empty()
// .top()

int main()
{

    // stack declaration
    stack<string> mystack;

    //pusing items in stack
    mystack.push("First");
    mystack.push("Second");
    mystack.push("Third");
    mystack.push("Fourth");

    //printing the top element, poping top till empty
    while (!mystack.empty())
    {
        cout << mystack.top() << endl;
        mystack.pop();
    }

    return 0;

    //Output (Depecting LIFO data structure):

    // Fourth
    // Third
    // Second
    // First
}

File: Python/DataStructure/Stack.py
class Stack(object):
    """
    Stack Implementation
    """
    def __init__(self):
        self.items = []
    
    def isEmpty(self):
        """Returns True if Stack is empty"""
        return self.items == []
    
    def push(self, item):
        """Adds item to the top of stack"""
        return self.items.append(item)

    def pop(self):
        """Pops out the top item"""
        return self.items.pop()
    
    def peek(self):
        """Peek the top item"""
        return self.items[-1]
    
    def size(self):
        """returns the size of stack"""
        return len(self.items)
